src/classlibnative/bcltype/objectnative.cpp

```diff
@@ -183,7 +183,7 @@ FCIMPL2(FC_BOOL_RET, ObjectNative::Equals, Object *pThisRef, Object *pCompareRef
     if (pThisMT != pCompareRef->GetMethodTable())
         FC_RETURN_BOOL(FALSE);
 
-    // Compare the contents (size - vtable - sink block index).
+    // Compare the contents (size - vtable - sync block index).
```

Comment:
They're two different words. A "sink" is a target or endpoint for something, a place where data end up. Here it's "sync" as in "synchronization".